Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java

```diff
@@ -1,22 +1,39 @@
 package dk.sdu.se.f23.InVasion.weapon;
 
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
 import dk.sdu.se.f23.InVasion.common.data.Entity;
 import dk.sdu.se.f23.InVasion.common.data.GameData;
 import dk.sdu.se.f23.InVasion.common.data.ProcessAt;
 import dk.sdu.se.f23.InVasion.common.data.World;
 import dk.sdu.se.f23.InVasion.common.services.EntityProcessingService;
 import dk.sdu.se.f23.InVasion.common.services.PluginService;
 
+import java.util.ArrayList;
+import java.util.Objects;
+
 public class WeaponPlugin implements PluginService {
     private String weaponName = "Steven";
+    private Texture texture = new Texture(Gdx.files.internal("Weapon/src/main/resources/TOWER.png"));
+    private int cost = 200;
 
 
     @Override
     public void onEnable(GameData data, World world) {
-        System.out.println("CREATED PLUGIN");
-        world.addWeapon(weaponName);
+        ArrayList<Object> weaponVariables = new ArrayList<>();
+        weaponVariables.add(weaponName);
```

Comment:
Make the name dynamic, based on what tower you have.

Reply:
i do not understand